Map<A, List<B>> or List<Map<A, B>> is not supported by introspection

### Steps to Reproduce

1. clone https://github.com/tchiotludo/micronaut-introspection
2. run unit test
3. see the stacktrace below
```
Caused by: java.lang.NoClassDefFoundError: E
	at bla.models.$Role$Introspection.getConstructorArguments(Unknown Source)
	at io.micronaut.jackson.modules.BeanIntrospectionModule$BeanIntrospectionDeserializerModifier.updateBuilder(BeanIntrospectionModule.java:277)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.buildBeanDeserializer(BeanDeserializerFactory.java:244)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.createBeanDeserializer(BeanDeserializerFactory.java:143)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer2(DeserializerCache.java:414)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer(DeserializerCache.java:349)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:264)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:491)
	at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:4669)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4448)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2696)
	at com.fasterxml.jackson.databind.ObjectMapper.treeToValue(ObjectMapper.java:3149)
	at io.micronaut.jackson.convert.JacksonConverterRegistrar.lambda$jsonNodeToObjectConverter$4(JacksonConverterRegistrar.java:208)
	at io.micronaut.core.convert.DefaultConversionService.convert(DefaultConversionService.java:135)
	at io.micronaut.context.env.DefaultEnvironment.convert(DefaultEnvironment.java:276)
	at io.micronaut.core.convert.ConversionService.convert(ConversionService.java:117)
	at io.micronaut.http.bind.binders.DefaultBodyAnnotationBinder.bind(DefaultBodyAnnotationBinder.java:78)
	at io.micronaut.http.bind.binders.DefaultBodyAnnotationBinder.bind(DefaultBodyAnnotationBinder.java:37)
	at io.micronaut.http.server.binding.RequestArgumentSatisfier.lambda$getValueForBlockingBodyArgumentBinder$3(RequestArgumentSatisfier.java:164)
	at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:274)
	at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:117)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildResultEmitter$11(RoutingInBoundHandler.java:1304)
	at io.reactivex.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:35)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at io.micronaut.http.server.context.ServerRequestContextFilter$1.subscribe(ServerRequestContextFilter.java:65)
	at io.reactivex.internal.operators.flowable.FlowableFromPublisher.subscribeActual(FlowableFromPublisher.java:29)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14868)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildExecutableRoute$5(RoutingInBoundHandler.java:1008)
	at io.micronaut.web.router.DefaultUriRouteMatch$1.execute(DefaultUriRouteMatch.java:81)
	at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:117)
	at io.micronaut.http.server.netty.RoutingInBoundHandler$1.executeRoute(RoutingInBoundHandler.java:941)
	at io.micronaut.http.server.netty.RoutingInBoundHandler$1.doOnComplete(RoutingInBoundHandler.java:934)
	at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71)
	at io.micronaut.http.server.netty.jackson.JsonContentProcessor$1.doOnComplete(JsonContentProcessor.java:140)
	at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.micronaut.core.async.processor.SingleThreadedBufferingProcessor.doOnComplete(SingleThreadedBufferingProcessor.java:48)
	at io.micronaut.jackson.parser.JacksonProcessor.doOnComplete(JacksonProcessor.java:133)
	at io.micronaut.core.async.subscriber.SingleThreadedBufferingSubscriber.onComplete(SingleThreadedBufferingSubscriber.java:70)
	at io.micronaut.http.server.netty.jackson.JsonContentProcessor.doOnComplete(JsonContentProcessor.java:165)
	at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71)
	at io.micronaut.http.netty.reactive.HandlerPublisher.complete(HandlerPublisher.java:498)
	at io.micronaut.http.netty.reactive.HandlerPublisher.handlerRemoved(HandlerPublisher.java:485)
	at io.netty.channel.AbstractChannelHandlerContext.callHandlerRemoved(AbstractChannelHandlerContext.java:979)
	at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:637)
	... 40 common frames omitted
Caused by: java.lang.ClassNotFoundException: E
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 89 common frames omitted
```

The bean are : 
```java
public class Role {
    @With
    private String id;

    @NotNull
    private String name;

    private String description;

    private Map<Permission, List<Action>> permissions;

    private List<Map<Permission, Action>> permissions2;

    @NotNull
    private boolean deleted = false;
}
```

Looking at the generated introspection : 
```java
    public Argument[] getConstructorArguments() {
        return new Argument[]{Argument.of(String.class, "arg0", (AnnotationMetadata)null, (Argument[])null), Argument.of(String.class, "arg1", (AnnotationMetadata)null, (Argument[])null), Argument.of(String.class, "arg2", (AnnotationMetadata)null, (Argument[])null), Argument.of(Map.class, "arg3", (AnnotationMetadata)null, new Argument[]{Argument.of(Permission.class, "K"), Argument.of(E.class, "V")}), Argument.of(List.class, "arg4", (AnnotationMetadata)null, new Argument[]{Argument.of(K.class, "E")})};
    }
```
where you can see `E.class`, `K.class`, ...

if I remove the `@Introspected` and the `@Valid` on the controller the test passed but I don't have anymore validation.

See on the micronaut **1.3.6**, but the git is on micronaut **2.0.0** that is also affected

--- a/inject-java-test/src/test/groovy/io/micronaut/inject/visitor/beans/BeanIntrospectionSpec.groovy
+++ b/inject-java-test/src/test/groovy/io/micronaut/inject/visitor/beans/BeanIntrospectionSpec.groovy
@@ -1601,6 +1601,36 @@ enum Test {
         thrown(InstantiationException)
     }
 
+    void "test constructor argument nested generics"() {
+        BeanIntrospection introspection = buildBeanIntrospection('test.Test', '''
+package test;
+
+import io.micronaut.core.annotation.Introspected;
+import java.util.List;
+import java.util.Map;
+
+@Introspected
+class Test {
+
+    public Test(Map<String, List<Action>> map) {
+    
+    }
+}
+
+class Action {
+
+}
+''')
+
+        expect:
+        introspection != null
+        introspection.constructorArguments[0].typeParameters.size() == 2
+        introspection.constructorArguments[0].typeParameters[0].typeName == 'java.lang.String'
+        introspection.constructorArguments[0].typeParameters[1].typeName == 'java.util.List<test.Action>'
+        introspection.constructorArguments[0].typeParameters[1].typeParameters.size() == 1
+        introspection.constructorArguments[0].typeParameters[1].typeParameters[0].typeName == 'test.Action'
+    }
+
     @Override
     protected JavaParser newJavaParser() {
         return new JavaParser() {

--- a/inject/src/main/java/io/micronaut/inject/writer/AbstractClassFileWriter.java
+++ b/inject/src/main/java/io/micronaut/inject/writer/AbstractClassFileWriter.java
@@ -473,7 +473,7 @@ public abstract class AbstractClassFileWriter implements Opcodes {
                 if (CollectionUtils.isNotEmpty(subArgs)) {
                     Map<String, Object> m = toTypeArguments(subArgs, visitedTypes);
                     if (CollectionUtils.isNotEmpty(m)) {
-                        map.put(entry.getKey(), m);
+                        map.put(entry.getKey(), Collections.singletonMap(getTypeForElement(ce), m));
                     } else {
                         map.put(entry.getKey(), Collections.singletonMap(entry.getKey(), className));
                     }

